core/templates/dev/head/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -21,22 +21,100 @@ require(
 require(
   'pages/exploration-editor-page/feedback-tab/services/' +
   'thread-status-display.service.ts');
+require(
+  'pages/exploration-editor-page/services/' +
+  'learner-answer-details-data.service.ts');
 require(
   'pages/exploration-editor-page/suggestion-modal-for-editor-view/' +
   'suggestion-modal-for-exploration-editor.service.ts');
 require('domain/utilities/UrlInterpolationService.ts');
+require('services/ExplorationHtmlFormatterService.ts');
 
 angular.module('oppia').factory('ImprovementModalService', [
   '$uibModal', 'AlertsService', 'ChangeListService', 'DateTimeFormatService',
-  'EditabilityService', 'ExplorationStatesService',
+  'EditabilityService', 'ExplorationHtmlFormatterService',
+  'ExplorationStatesService', 'LearnerAnswerDetailsDataService',
   'SuggestionModalForExplorationEditorService', 'ThreadDataService',
   'ThreadStatusDisplayService', 'UrlInterpolationService', 'UserService',
   function(
       $uibModal, AlertsService, ChangeListService, DateTimeFormatService,
-      EditabilityService, ExplorationStatesService,
+      EditabilityService, ExplorationHtmlFormatterService,
+      ExplorationStatesService, LearnerAnswerDetailsDataService,
       SuggestionModalForExplorationEditorService, ThreadDataService,
       ThreadStatusDisplayService, UrlInterpolationService, UserService) {
     return {
+      openLearnerAnswerDetails: function(learnerAnswerDetails) {
+        return $uibModal.open({
+          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+            '/pages/exploration-editor-page/improvements-tab/templates/' +
+            'answer-details-modal.template.html'),
+          resolve: {
+            isUserLoggedIn: function() {
+              return UserService.getUserInfoAsync().then(function(userInfo) {
+                return userInfo.isLoggedIn();
+              });
+            },
+          },
+          controller: [
+            '$scope', '$uibModalInstance', 'isUserLoggedIn',
+            function($scope, $uibModalInstance, isUserLoggedIn) {
+              $scope.selectedLearnerAnswerInfo = [];
+              $scope.learnerAnswerDetails = learnerAnswerDetails;
+              $scope.currentLearnerAnswerInfo = null;
+              $scope.viewAnswerDetails = false;
+              $scope.changeView = function(learnerAnswerInfo) {
+                $scope.currentLearnerAnswerInfo = learnerAnswerInfo;
+                $scope.viewAnswerDetails = !($scope.viewAnswerDetails);
+              };
+              $scope.getLocaleAbbreviatedDatetimeString = (
+                DateTimeFormatService.getLocaleAbbreviatedDatetimeString);
+              $scope.getLearnerAnswerInfos = function() {
+                return $scope.learnerAnswerDetails.learnerAnswerInfoData;
+              };
+              $scope.getAnswerDetails = function(learnerAnswerInfo) {
+                return learnerAnswerInfo._answerDetails;
```

Comment:
Use `getAnswerDetails()` instead of using the field directly.

Reply:
Done